flink-runtime/src/main/java/org/apache/flink/runtime/state/DefaultOperatorStateBackend.java

```diff
@@ -185,13 +185,18 @@ public void dispose() {
 				new HashMap<>(registeredStates.size());
 
 		// eagerly create deep copies of the list states in the sync phase, so that we can use them in the async writing
-		for (Map.Entry<String, PartitionableListState<?>> entry : this.registeredStates.entrySet()) {
-
-			PartitionableListState<?> listState = entry.getValue();
-			if (null != listState) {
-				listState = listState.deepCopy();
+		ClassLoader snapshotClassLoader = Thread.currentThread().getContextClassLoader();
+		try {
+			Thread.currentThread().setContextClassLoader(userClassloader);
```

Comment:
(nitpick) `Thread.currentThread().setContextClassLoader(userClassloader)` should occur before the `try` block is entered.   Setting the TCCL occurs in about a half-dozen places in Flink, and this is inconsistent with most of them.